core/block.go

```diff
@@ -149,3 +150,72 @@ func (b *Block) post07Hash() *felt.Felt {
 		b.ParentHash,            // parent block hash
 	)
 }
+
+// TransactionCommitment is the root of a height 64 binary Merkle Patricia tree of the
+// transaction hashes and signatures in a block.
+func TransactionCommitment(transactions []*TransactionReceipt) (*felt.Felt, error) {
+	var transactionCommitment *felt.Felt
+	trie.RunOnTempTrie(64, func(trie *trie.Trie) error {
+		zeroFelt := new(felt.Felt)
+		emptySignatureHash := crypto.Pedersen(zeroFelt, zeroFelt)
+		for i, transaction := range transactions {
+			var signaturesHash *felt.Felt
+			if transaction.Type == Invoke {
+				signaturesHash = crypto.PedersenArray(transaction.Signatures...)
+			} else {
+				signaturesHash = emptySignatureHash
+			}
+			transactionAndSignatureHash := crypto.Pedersen(transaction.TransactionHash, signaturesHash)
+			err := trie.Put(new(felt.Felt).SetUint64(uint64(i)), transactionAndSignatureHash)
+			if err != nil {
+				return err
+			}
+		}
+		root, err := trie.Root()
+		if err != nil {
+			return err
+		}
+		transactionCommitment = root
+		return nil
+	})
+
+	return transactionCommitment, nil
+}
+
+type EventData struct {
+	EventCommitment *felt.Felt // root of a height 64 binary Merkle Patricia tree of the events in a block.
+	EventCount      *felt.Felt // number of events in a block.
+}
+
+// ComputeEventData computes the event commitment and event count for a block.
+func ComputeEventData(receipts []*TransactionReceipt) (*EventData, error) {
```

Comment:
Rather than creating a new `EventData` type solely for the purpose of combining return values from this function, it may be better to return `(*felt.Felt, uint64, error)`.